### ISSUE TITLE
Deep Planets refusing to surrender due to fleeing defensive ship

### ISSUE THREAD

**Issue by MCOfficer** [endless-sky/endless-sky#2430]
i tried to dominate Munnin Station ("The Gate to the Deep") several times, but they simply refuse to surrender. There are no Defense Ships left in the system, i even destroyed the disabled ones, but no matter what i try, the station just tells me that they are "not ready to surrender yet".
i tried to dominate Huggin Sation in the meantime, and it worked like a charm.

attached is a video showing 2 attempts and the successful Huggin Station domination. i had to speed it up a lot, but that messed up the sound, so you better mute it.
https://www.dropbox.com/s/b59cjpzljudq2xv/MunninStation2.mp4

**Comment by lheckemann** [comment]
Are you sure there's not still a defence fleet missile ship that's keeping its distance from your fleet and so avoiding destruction?

**Comment by MCOfficer** [comment]
i was spamming r every now and then. if there had been a hostile ship of any sort, it would've appeared on my scanners. the system was clean, i am certain.

**Comment by KiLEdEnNis** [comment]
I know this issue (not sure if it's a duplicate), and it only happens in the Deep: I think I once caught a defense ship fleeing the system, not 100% sure but I think that's the issue, and I'm not sure but I think the Deep had an unarmed ship in a fleet?

**Comment by MCOfficer** [comment]
that is a good point, i remember a flivver leaving the system.

looking into the data files, there is indeed a flivver without weapons (the racing variant) that spawns as part of the "Deep Small" Defensive fleet.

**Comment by MCOfficer** [comment]
that is a good point, i remember a flivver leaving the system.

looking into the data files, there is indeed a flivver without weapons (the racing variant) that spawns as part of the "Deep Small" Defensive fleet.

**Comment by KiLEdEnNis** [comment]
delete it in your data.txt & try & reproduce the bug a few times
it'll take a while but if it doesn't happen in ~3 more tries it may have been the source
that or instead delete everything else & tribute & watch if any Flivvers leave the system

**Comment by endless-sky** [comment]
The real bug is why a defense ship, which ought to have the "staying" personality, would ever leave the system, and I can't seem to figure out how that is happening. And, I've tested a defense fleet of all unarmed Flivvers, and I can't get them to flee the system, even if I hit them enough to bring down their shields and damage their hulls.

**Comment by MCOfficer** [comment]
so, i'll just crosspost this video, in case you haven't seen it. this was some testing with loads of unarmed flivvers as defensive fleet, and the results are pretty clear.
https://www.dropbox.com/s/k412hno94vbx67c/2017-04-20%2015-57-20.mp4

**Comment by endless-sky** [comment]
The bug with defense ships fleeing is fixed in e3cb540. The defense fleets should also be updated to not include non-military ships.

**Comment by EndrosG** [comment]
@endless-sky : the Flivver was removed by the user, see pull #2431

**Comment by endless-sky** [comment]
Pull #2431 doesn't work, because that would completely eliminate racing Flivvers throughout Deep space.

**Comment by MCOfficer** [comment]
True that. latest commit should fix it, by using a defensive and a normal "small deep" fleet.

**Comment by EndrosG** [comment]
From what I checked when it was created, the Flivver was still in Northern Merchants fleet and all systems which use Deep Fleets also use Northern Merchants fleets. So I thought this problem would not arise.

**Comment by EndrosG** [comment]
```
system "Epsilon Leonis"
	fleet "Small Northern Merchants" 400
	fleet "Large Northern Merchants" 1000
	fleet "Small Deep" 1000
	fleet "Large Deep" 3000

fleet "Small Northern Merchants"
	variant 3
		"Flivver"
	variant 2
		"Flivver (Racing)"
```

**Comment by endless-sky** [comment]
Fixed in f0678b3.
